packages/mui-base/src/useInput/useInput.test.tsx

```diff
@@ -20,4 +22,29 @@ describe('useInput', () => {
       expect(inputRef.current).to.deep.equal(getByRole('textbox'));
     });
   });
+
+  describe('prop: disabled', () => {
+    it('should reset the focused state if getting disabled', () => {
```

Comment:
The code related to this used to be in InputBase [here](https://github.com/mui/material-ui/blob/master/packages/mui-material/src/InputBase/InputBase.js#L336-L345) but it's now implemented in `useInput`, so I figure it's probably better to test it here? @michaldudak

Reply:
Yeah, let's test the hooks extensively so that other products can just assume they work correctly and focus on testing only the additional functionality they provide.

Reply:
Sounds good – I'll add a task to port additional relevant tests to Base in the [issue](https://github.com/mui/material-ui/issues/38372)